pkg/querier/http.go

```diff
@@ -58,44 +59,43 @@ func directionParam(values url.Values, name string, def logproto.Direction) (log
 	return logproto.Direction(d), nil
 }
 
-// QueryHandler is a http.HandlerFunc for queries.
-func (q *Querier) QueryHandler(w http.ResponseWriter, r *http.Request) {
-	params := r.URL.Query()
-	query := params.Get("query")
+func httpRequestToQueryRequest(httpRequest *http.Request) (logproto.QueryRequest, error) {
+	params := httpRequest.URL.Query()
+	now := time.Now()
+	queryRequest := logproto.QueryRequest{
+		Regex: params.Get("regexp"),
+	}
+
+	queryRequest.Query = params.Get("query")
 	limit, err := intParam(params, "limit", defaultQueryLimit)
 	if err != nil {
-		http.Error(w, err.Error(), http.StatusBadRequest)
-		return
+		return queryRequest, err
```

Comment:
When there is an error, best to return `nil, err`

Reply:
(You need to make the return type *logproto.QueryRequest)

Reply:
Actually I did this deliberately because returning pointer would escape it to heap and stay there until WS is open.
I will handle it in caller to pass on the copy down the call to get it GCed.